extra/mariabackup/xtrabackup.cc

```diff
@@ -5508,10 +5508,23 @@ xtrabackup_apply_delta(
 					buf + FSP_HEADER_OFFSET + FSP_SIZE);
 				if (mach_read_from_4(buf
 						     + FIL_PAGE_SPACE_ID)) {
-					if (!os_file_set_size(
-						    dst_path, dst_file,
-						    n_pages * page_size))
+#ifdef _WIN32
+					os_offset_t last_page =
+					  os_file_get_size(dst_file) /
+					  page_size;
+
+					if (last_page < n_pages &&
+					    !os_file_set_size(
+					       dst_path, dst_file,
+					       n_pages * page_size))
+						goto error;
+#else
+					if (os_file_set_size(
+					      dst_path, dst_file,
+					      n_pages *page_size))
 						goto error;
+#endif /* _WIN32 */
```

Comment:
Can you please minimize the diff as follows:
* Retain the original formatting of the non-Windows code
* Do not add `goto error`
* Add the `#endif` before the existing `goto error` so that it will be used in the Windows specific code path.

I would also suggest adding a source code comment that explains that only on Windows, `os_file_set_size()` would be able to shrink the size of a file.